src/datanode/src/instance/sql.rs

```diff
@@ -41,10 +41,30 @@ impl Instance {
     async fn do_execute_sql(&self, stmt: Statement, query_ctx: QueryContextRef) -> Result<Output> {
         match stmt {
             Statement::Insert(insert) => {
-                let request =
-                    SqlHandler::insert_to_request(self.catalog_manager.clone(), *insert, query_ctx)
-                        .await?;
-                self.sql_handler.insert(request).await
+                let request = SqlHandler::insert_to_request(
+                    self.catalog_manager.clone(),
+                    &insert,
+                    query_ctx.clone(),
+                )
+                .await?;
+                if let Some(request) = request {
+                    self.sql_handler.insert(request).await
+                } else {
+                    // Can't cast the stmt into request, let's execute it by query engine.
```

Comment:
If `do_execute_sql` is invoked with insert stmt, it has to be an `Insert into ... values ...`. Otherwise the stmt is intercepted in `src/frontend/src/statement.rs`.

Reply:
Ok, so i don't have to check the request here.